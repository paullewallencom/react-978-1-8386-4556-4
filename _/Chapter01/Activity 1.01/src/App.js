import React, { Component } from "react";

const Header = props => <h1>{props.title}</h1>;

const InventoryItem = props => (
  <div className="InventoryItem">
    <strong>{props.itemName}</strong>
    <hr />
    <p>{props.itemPrice}</p>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [
        { itemName: "Shoe", itemPrice: 5 },
        { itemName: "Sock", itemPrice: 3 }
      ]
    };
  }
  render() {
    return (
      <div className="App">
        <Header title="BuyStuff" />
        <InventoryItem
          itemName={this.state.items[0].itemName}
          itemPrice={this.state.items[0].itemPrice}
        />
        <InventoryItem
          itemName={this.state.items[1].itemName}
          itemPrice={this.state.items[1].itemPrice}
        />
      </div>
    );
  }
}

export default App;
