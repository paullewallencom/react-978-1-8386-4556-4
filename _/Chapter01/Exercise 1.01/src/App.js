import React from "react";

class App extends React.Component {
  render() {
    return (
      <div className="Example" id="my-element">
        <div
          className="greeting"
          style={{ background: "black", color: "white" }}
          onClick={() => alert("hello")}
        >
          Hello World
        </div>
      </div>
    );
  }
}

export default App;
