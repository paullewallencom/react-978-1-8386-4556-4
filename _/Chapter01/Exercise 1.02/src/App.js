import React from "react";

const App = props => (
  <div>
    <h1>My App</h1>
    <br />
    Hello React!
  </div>
);

export default App;
