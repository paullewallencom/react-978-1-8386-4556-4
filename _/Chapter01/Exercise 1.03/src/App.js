import React, { Component } from 'react';
 
const Greeting = props => <p>Hello {props.name}!</p>;
 
class App extends Component {
  constructor(props) {
    super(props);
    this.title = 'React App';
    this.state = { clickCounter: 0 };
  }
  renderClickCount() {
    return <p>I've been clicked {this.state.clickCounter} times!</p>;
  }
  render() {
    return (
      <div>
        <h1>{this.title}</h1>
        <br />
        <Greeting name="User" />
        <br />
        {this.renderClickCount()}
      </div>
    );
  }
}
 
export default App;

