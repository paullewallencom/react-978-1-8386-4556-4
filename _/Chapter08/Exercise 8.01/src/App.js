import React from 'react';
import {UncontrolledForm} from './UncontrolledForm';
import './App.css';

function App() {
  return (
  <div className="App">
    <UncontrolledForm/>
  </div>
  );
}
export default App; 
