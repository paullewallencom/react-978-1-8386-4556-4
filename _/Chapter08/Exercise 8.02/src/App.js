import React from 'react';
import {ControlledForm} from './ControlledForm';
import './App.css';

function App() {
  return (
  <div className="App">
  <ControlledForm />
  </div>
  );
}
export default App;

