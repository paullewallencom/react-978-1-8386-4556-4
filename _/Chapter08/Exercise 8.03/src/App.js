import React, { Component } from 'react';
import {MyEnhancedLoginForm} from "./WithFormikExample";
import './App.css';

class App extends Component {

  render() {
    // And finally, our render method. Nothing too terribly interesting here.
    return (
      <div className="App">
        <MyEnhancedLoginForm />
      </div>
    );
  }
}

export default App;

