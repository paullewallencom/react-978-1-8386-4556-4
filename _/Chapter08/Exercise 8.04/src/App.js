import React from "react";
import "./App.css";
import FormValidationControl from "./FormYupValidation";

function App() {
  return (
    <div className="App">
      <FormValidationControl />
    </div>
  );
}

export default App;
