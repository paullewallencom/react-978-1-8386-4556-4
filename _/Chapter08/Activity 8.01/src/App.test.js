import React from 'react';
import {render, fireEvent, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import App from './App';

test('renders correctly', () => {
  const {container} = render(<App/>);
  expect(container).toMatchSnapshot();
});

test('loads and displays form', async () => {
  //Arrange
  const { container } = render(<App />);
  const tos = container.querySelector('input[type="checkbox"]');
  const name = container.querySelector('input[type="text"]');
  const email = container.querySelector('input[type="email"]');
  const passwords = container.querySelectorAll('input[type="password"]');
  const password = passwords[0];
  const passwordConfirm = passwords[1];

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: 'example@email.com'
      }
    });
    fireEvent.change(password, {
      target: {
        value: 'password'
      }
    });
    fireEvent.change(passwordConfirm, {
      target: {
        value: 'password'
      }
    });
    fireEvent.change(name, {
      target: {
        value: 'Theo'
      }
    });
    fireEvent.change(tos, {
      target: {
        checked: true
      }
    })
  });

  //Assert
  expect(email.value).toBe('example@email.com');
  expect(password.value).toBe('password');
  expect(passwordConfirm.value).toBe('password');
  expect(name.value).toBe('Theo');
  expect(tos.checked).toBeTruthy()
});

test('displays an error message', async () => {
  //Arrange
  const { container } = render(<App />);
  const tos = container.querySelector('input[type="checkbox"]');
  const name = container.querySelector('input[type="text"]');
  const email = container.querySelector('input[type="email"]');
  const passwords = container.querySelectorAll('input[type="password"]');
  const submit = container.querySelector('input[type="submit"]');
  const password = passwords[0];
  const passwordConfirm = passwords[1];

  //Act
  await wait(() => {
    fireEvent.change(email, {
      target: {
        value: ''
      }
    });
    fireEvent.change(password, {
      target: {
        value: ''
      }
    });
    fireEvent.change(passwordConfirm, {
      target: {
        value: 'aa'
      }
    });
    fireEvent.change(name, {
      target: {
        value: ''
      }
    });
    fireEvent.change(tos, {
      target: {
        value: true
      }
    });

    fireEvent.blur(email);
    fireEvent.blur(password);
    fireEvent.blur(passwordConfirm);
    fireEvent.blur(name);
  });

  //Assert
  expect(container).toMatchSnapshot();
});
