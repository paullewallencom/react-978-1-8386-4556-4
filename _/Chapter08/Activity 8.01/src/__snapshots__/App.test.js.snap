// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays an error message 1`] = `
<div>
  <div
    class="App"
  >
    <form>
      <label>
        Name:
      </label>
      <br />
      <input
        name="name"
        type="text"
        value=""
      />
      <br />
      <label>
        Password:
      </label>
      <br />
      <input
        name="password"
        type="password"
        value=""
      />
      <br />
      <label>
        Password Match:
      </label>
      <br />
      <input
        name="passwordMatch"
        type="password"
        value="aa"
      />
      <br />
      <label>
        Email:
      </label>
      <br />
      <input
        name="email"
        type="email"
        value=""
      />
      <br />
      <label>
        Accept Terms and Conditions:
      </label>
      <input
        name="acceptTAC"
        type="checkbox"
        value="false"
      />
      <br />
      <br />
      <input
        type="submit"
        value="Register"
      />
    </form>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <form>
      <label>
        Name:
      </label>
      <br />
      <input
        name="name"
        type="text"
        value=""
      />
      <br />
      <label>
        Password:
      </label>
      <br />
      <input
        name="password"
        type="password"
        value=""
      />
      <br />
      <label>
        Password Match:
      </label>
      <br />
      <input
        name="passwordMatch"
        type="password"
        value=""
      />
      <br />
      <label>
        Email:
      </label>
      <br />
      <input
        name="email"
        type="email"
        value=""
      />
      <br />
      <label>
        Accept Terms and Conditions:
      </label>
      <input
        name="acceptTAC"
        type="checkbox"
        value="false"
      />
      <br />
      <br />
      <input
        type="submit"
        value="Register"
      />
    </form>
  </div>
</div>
`;
